instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/logging/LogbackAppenderApplicationListener.java

```diff
@@ -53,35 +54,129 @@ private static boolean isAssignableFrom(Class<?> type, Class<?>... supportedType
   @Override
   public void onApplicationEvent(ApplicationEvent event) {
     if (event instanceof ApplicationEnvironmentPreparedEvent // Event for which
-        // org.springframework.boot.context.logging.LoggingApplicationListener
-        // initializes logging
-        && !isOpenTelemetryAppenderAlreadyConfigured()) {
-      ch.qos.logback.classic.Logger logger =
-          (ch.qos.logback.classic.Logger)
-              LoggerFactory.getILoggerFactory().getLogger(Logger.ROOT_LOGGER_NAME);
-
-      OpenTelemetryAppender appender = new OpenTelemetryAppender();
-      appender.start();
-      logger.addAppender(appender);
+    // org.springframework.boot.context.logging.LoggingApplicationListener
+    // initializes logging
+    ) {
+      Optional<OpenTelemetryAppender> existingOpenTelemetryAppender = findOpenTelemetryAppender();
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent =
+          (ApplicationEnvironmentPreparedEvent) event;
+      if (existingOpenTelemetryAppender.isPresent()) {
+        reInitializeOpenTelemetryAppender(
+            existingOpenTelemetryAppender, applicationEnvironmentPreparedEvent);
+      } else {
+        addOpenTelemetryAppender(applicationEnvironmentPreparedEvent);
+      }
+    }
+  }
+
+  private static void reInitializeOpenTelemetryAppender(
+      Optional<OpenTelemetryAppender> existingOpenTelemetryAppender,
+      ApplicationEnvironmentPreparedEvent applicationEnvironmentPreparedEvent) {
+    OpenTelemetryAppender openTelemetryAppender = existingOpenTelemetryAppender.get();
+    openTelemetryAppender.stop(); // The OpenTelemetry appender is stopped and restarted from the
+    // org.springframework.boot.context.logging.LoggingApplicationListener.initialize
+    // method
+    // The OpenTelemetryAppender initializes the LoggingEventMapper in the start() method. So, here
+    // we stop the OpenTelemetry appender before its re-initialization and its restart.
```

Comment:
might be clearer to move this multiline comment above the line?

Reply:
Done